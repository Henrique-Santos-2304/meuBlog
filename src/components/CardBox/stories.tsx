import { Story, Meta } from "@storybook/react";
import CarBox from ".";

export default {
  title: "CarBox",
  component: CarBox,
} as Meta;
const props = {
  url: "string",
  title: "string",
  subDescription: "string",
};
export const Basic: Story = (args) => <CarBox {...args} {...props} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <CarBox {...args} {...props} />;
