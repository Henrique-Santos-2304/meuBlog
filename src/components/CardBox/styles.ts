import styled from "styled-components";

export const Wrapper = styled.div`
  position: absolute;
  bottom: 1.4rem;
  left: 50%;
  transform: translateX(-50%);

  button {
    svg {
      margin-bottom: 0.3rem;
    }
  }
`;
