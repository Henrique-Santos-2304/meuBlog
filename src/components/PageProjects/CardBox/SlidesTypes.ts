type buttonProps = {
  __typename?: any;
  url: string;
  label: string;
};
export type imageProps = {
  url: string;
  title: string;
  subDescription: string;
  alt: string;
  buttons: buttonProps[];
};
