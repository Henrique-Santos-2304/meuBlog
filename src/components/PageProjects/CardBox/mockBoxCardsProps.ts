import { mockButtons } from "../Buttons/mockButtonsProps";

export const mockTypesBoxCard = {
  url: "img.png",
  title: "projeto 1",
  subDescription: "descrição do projeto",
  alt: "um mini Slide",
  buttons: [...mockButtons.buttons],
};
