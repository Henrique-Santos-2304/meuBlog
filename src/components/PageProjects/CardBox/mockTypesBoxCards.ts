export const mockTypesBoxCard = {
  url: "img.png",
  title: "minha foto",
  subDescription: "description ",
  alt: "um mini Slide",
  buttons: [
    {
      url: "codigo.com",
      label: "ver projeto",
    },
    {
      url: "codigo.com",
      label: "ver c√≥digo",
    },
  ],
};
