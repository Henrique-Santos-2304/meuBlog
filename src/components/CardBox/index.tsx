import {
  Box,
  Image,
  Text,
  Button,
  ButtonGroup,
  Flex,
  Link,
  useDisclosure,
} from "@chakra-ui/react";
import Headings from "components/Headings";
import { FaGithub } from "react-icons/fa";
import { ModalCardBox } from "./ModalCardBox";
import * as S from "./styles";

type buttonProps = {
  __typename?: any;
  url: string;
  label: string;
};
type imageProps = {
  url: string;
  title: string;
  subDescription: string;
  alt: string;
  buttons: buttonProps[];
};

const CardBox = ({ url, title, subDescription, alt, buttons }: imageProps) => {
  const { isOpen, onOpen, onClose } = useDisclosure();

  return (
    <Box
      h={["40rem", "40rem", "42rem", "48rem"]}
      bg="blackAlpha.700"
      p={["2rem", "6rem"]}
      mx={["0.5rem", "1rem"]}
      position="relative"
      borderRadius="3.5rem"
    >
      <Flex
        position="absolute"
        top="-1.2rem"
        left="50%"
        transform="translate(-50%)"
        borderRadius="4rem"
      >
        <Headings
          msg={title}
          color="orange"
          margins="1rem auto"
          fontSize="2rem"
        />
      </Flex>
      <Image
        src={url}
        borderRadius="3rem"
        boxShadow="0 0 2rem white, 0 0 1rem orange"
        w="100%"
        h={["70%", "100%"]}
        mt={["4rem", "0"]}
        alt={alt}
      />
      <S.Wrapper>
        <ButtonGroup spacing={["20px", "25px", "40px", "80px"]}>
          <Button
            colorScheme="green"
            borderRadius="0.7rem"
            textAlign="center"
            p={["1.2rem 2rem", "1.6rem 2.5rem"]}
            fontSize={["1.2rem", null, null, "1.8rem"]}
            _hover={{ border: "0.1rem solid white" }}
            onClick={onOpen}
          >
            {buttons[0].label}
          </Button>
          <Link href={buttons[1].url} isExternal>
            <Button
              bg="blackAlpha.600"
              borderRadius="0.7rem"
              color="gray.100"
              p={["1rem 2rem", "1.6rem 2.5rem"]}
              _hover={{ border: "0.1rem solid white" }}
              fontSize={["1.2rem", null, null, "1.8rem"]}
            >
              <Flex justify="center" align="center">
                <FaGithub />
                <Text ml="0.9rem">{buttons[1].label}</Text>
              </Flex>
            </Button>
          </Link>
        </ButtonGroup>
      </S.Wrapper>
      <ModalCardBox
        isOpen={isOpen}
        onClose={onClose}
        subDescription={subDescription}
        title={title}
        url={url}
      />
    </Box>
  );
};

export default CardBox;
