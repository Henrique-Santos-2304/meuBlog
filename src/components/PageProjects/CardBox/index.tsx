import * as S from "./styles";
import { Box, Image, Flex, useDisclosure } from "@chakra-ui/react";

import { ModalCardBox } from "../ModalCardBox";
import Headings from "components/globalComponents/Headings";
import Buttons from "components/PageProjects/Buttons";
import { imageCardProps } from "graphql/typesFinal/queryProjects";

const CardBox = ({
  url,
  title,
  subDescription,
  alt,
  buttons,
}: imageCardProps) => {
  const { isOpen, onOpen, onClose } = useDisclosure();

  return (
    <Box
      h={["35rem", "40rem", "44rem", "46rem"]}
      bg="blackAlpha.700"
      p={["3rem", "7rem"]}
      mx={["0.5rem", "1rem"]}
      position="relative"
      borderRadius="3.5rem"
      aria-label="cardsSlides"
    >
      <Flex
        position="absolute"
        top="-1.2rem"
        left="50%"
        transform="translate(-50%)"
        borderRadius="4rem"
      >
        <Headings msg={title} color="gray.100" fontSize="2rem" />
      </Flex>
      <Image
        src={url}
        borderRadius="3rem"
        boxShadow="0 0 2rem white, 0 0 1rem orange"
        w={["90%", "100%"]}
        h={["60%", "100%"]}
        mx="auto"
        mt={["4rem", "0"]}
        alt={alt}
      />
      <S.Wrapper>
        <Buttons buttons={buttons} onOpen={onOpen} />
      </S.Wrapper>

      <ModalCardBox
        isOpen={isOpen}
        onClose={onClose}
        subDescription={subDescription}
        title={title}
        url={url}
      />
    </Box>
  );
};

export default CardBox;
