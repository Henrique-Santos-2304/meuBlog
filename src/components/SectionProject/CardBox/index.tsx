import * as S from "./styles";
import Link from "next/link";

type cardsSlides = {
  title: string;
  img: string;
  buttonCode: string;
  marginBox: string;
};
const CardBox = ({ title, img, buttonCode, marginBox }: cardsSlides) => (
  <S.ContentSection marginBox={marginBox} aria-label="Titulo do projeto">
    <S.ContentTitle>
      <S.TitleProject>{title}</S.TitleProject>
    </S.ContentTitle>
    <S.Wrapper aria-label="Imagem do projeto">
      <S.PhotoProject src={img} layout="fill" quality={100} />
    </S.Wrapper>
    <S.ContentButton>
      <Link href={buttonCode} passHref>
        <a target="_blank">
          <S.ButtonCode aria-label="Link para código do Projeto">
            <S.Icon />
            Ver Código
          </S.ButtonCode>
        </a>
      </Link>
    </S.ContentButton>
  </S.ContentSection>
);

export default CardBox;
