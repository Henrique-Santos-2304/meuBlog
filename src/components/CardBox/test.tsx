import CardBox from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
const props = {
  url: "test.jpg",
  title: "minha foto",
  subDescription: "description",
};

describe("<CarBox />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<CardBox {...props} />);
    expect(container).toMatchSnapshot();
  });
  it("should have a image on Card ", () => {
    renderTheme(<CardBox {...props} />);
    expect(screen.getByRole("img")).toBeInTheDocument();
  });
  it("should have a title on Card ", () => {
    renderTheme(<CardBox {...props} />);
    expect(
      screen.getByRole("heading", { name: /minha foto/i })
    ).toBeInTheDocument();
  });
  it("should have a description project on Card ", () => {
    renderTheme(<CardBox {...props} />);
    expect(screen.getByText(/description/i)).toBeInTheDocument();
  });
  it("should have a button to view full project ", () => {
    renderTheme(<CardBox {...props} />);
    expect(
      screen.getByRole("button", { name: /ver projeto/i })
    ).toBeInTheDocument();
  });
  it("should have a button to view a code of the project ", () => {
    renderTheme(<CardBox {...props} />);
    expect(
      screen.getByRole("button", { name: /ver c√≥digo/i })
    ).toBeInTheDocument();
  });
});
