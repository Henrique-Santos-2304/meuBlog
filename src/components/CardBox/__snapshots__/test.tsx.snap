// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarBox /> should render  1`] = `
.c1 {
  color: #E5E7EB;
  font-size: 2.0rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  color: #06092b;
  font-size: 1.4rem;
  position: absolute;
  bottom: 2rem;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  overflow: hidden;
}

<div>
   
  <div
    class="css-1sz7gil"
  >
    <img
      class="chakra-image css-14lr2s0"
      src="test.jpg"
    />
    <h2
      class="chakra-heading css-vqu1z5"
    >
      minha foto
    </h2>
    <p
      class="chakra-text css-23t00v"
    >
      description
    </p>
    <div
      class="c0"
    >
      <div
        class="chakra-button__group css-gg6mva"
        role="group"
      >
        <button
          class="chakra-button css-10tw6y9"
          type="button"
        >
          Ver projeto
        </button>
        <button
          class="chakra-button css-1hw5e2v"
          type="button"
        >
          <span
            class="chakra-button__icon css-1wh2kri"
          >
            <div
              aria-hidden="true"
              class="c1"
              focusable="false"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 496 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                />
              </svg>
            </div>
          </span>
          Ver CÃ³digo
        </button>
      </div>
    </div>
  </div>
</div>
`;
